test/integrationtests/step_impl/test_helpers.py

```
@@ -0,0 +1,66 @@
+import time
+
+import boto3
+import botocore
+from getgauge.python import Messages
+
+from .tf_aws_resources import get_terraform_output
+from utils import get_datetime_now
+
+REGION_NAME = "eu-west-2"
+dynamodb = boto3.resource("dynamodb", REGION_NAME)
+s3 = boto3.client("s3", REGION_NAME)
+stepfunctions = boto3.client("stepfunctions", REGION_NAME)
+
+
+def await_s3_object_exists(bucket, key):
+    try:
+        waiter = s3.get_waiter("object_exists")
+        waiter.wait(Bucket=bucket, Key=key)
+        Messages.write_message(f"Object exists: {bucket}/{key}")
+        return True
+    except botocore.exceptions.WaiterError:
+        Messages.write_message(f"Object not found after waiting: {bucket}/{key}")
+        return False
+
+
+def get_latest_jobid():
+    job_table = dynamodb.Table(get_terraform_output("jobs_table"))
+    job_data = job_table.scan()
+    job_items = []
+    for key, value in job_data.items():
+        if key == "Items":
+            job_items = [j for j in value]
+            job_items = sorted(job_items, reverse=True, key=lambda i: i["Timestamp"])
+            if job_items:
+                latest_job_id = job_items[0]
+                return latest_job_id["Id"]
+
+
+def create_timestamp():
+    now = get_datetime_now().now()
+    return "%02d%02d%02d%02d%d%d" % (
+        now.year,
+        now.month,
+        now.day,
+        now.minute,
+        now.second,
+        now.microsecond,
+    )
+
+
+def await_stepfunction_succeeded(execution_arn, poll_limit=5, poll_count=0, poll_sleep=5):
+    while True:
+        describe_execution = stepfunctions.describe_execution(executionArn=execution_arn)
+        status = describe_execution["status"]
+
+        if status == "SUCCEEDED":
+            return describe_execution
+        else:
+            poll_count += 1
+
+        if poll_count >= poll_limit:
+            Messages.write_message(f"LR 10 StepFunction arn: {execution_arn} status: {status}")
+            return describe_execution
```

Maybe raise an exception from this since it would be a fail for it not to succeed

---

Discussed with James to leave as is as it's consistent with the test code style